rust/pattern/variable/builder/expression.rs

```diff
@@ -21,91 +21,71 @@
  */
 
 use crate::{
-    common::token::{Function as FunctionToken, Operation as OperationToken},
-    pattern::expression::{Expression, Function, Operation},
+    common::token::ArithmeticOperator,
+    pattern::expression::{Expression, Operation},
 };
 
+pub trait SubExpression {
+    fn into_expression(self) -> Expression;
+}
```

Comment:
There is a simpler way:
```suggestion
pub trait SubExpression: Into<Expression> {}
```